sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -52,105 +57,142 @@ public HttpLoggingPolicy(HttpLogOptions httpLogOptions) {
+            this.allowedHeaderNames = httpLogOptions.getAllowedHeaderNames()
+                .stream()
+                .map(String::toLowerCase)
+                .collect(Collectors.toSet());
+            this.allowedQueryParameterNames = httpLogOptions.getAllowedQueryParamNames()
+                .stream()
+                .map(String::toLowerCase)
+                .collect(Collectors.toSet());
+        }
     }
 
     @Override
     public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
-        //
-        if (httpLogOptions == null || httpLogOptions.getLogLevel() == HttpLogDetailLevel.NONE) {
-            // ideally, httpLogOptions should not be null but adding a null check now will be a breaking change
-            // nothing to log, continue with the next pipeline policy, if there is one
-            // "next" cannot be null here
+        // No logging will be performed, trigger a no-op.
+        if (httpLogDetailLevel == HttpLogDetailLevel.NONE) {
             return next.process();
         }
 
-        Optional<Object> data = context.getData("caller-method");
-        String callerMethod = (String) data.orElse("");
-
-        final ClientLogger logger = new ClientLogger(callerMethod);
+        final ClientLogger logger = new ClientLogger((String) context.getData("caller-method").orElse(""));
         final long startNs = System.nanoTime();
-        Mono<Void> logRequest = logRequest(logger, context.getHttpRequest());
-        Function<HttpResponse, Mono<HttpResponse>> logResponseDelegate =
-            logResponseDelegate(logger, context.getHttpRequest().getUrl(), startNs);
-        return logRequest.then(next.process()).flatMap(logResponseDelegate)
+
+        return logRequest(logger, context.getHttpRequest())
+            .then(next.process())
+            .flatMap(response -> logResponse(logger, response, startNs))
             .doOnError(throwable -> logger.warning("<-- HTTP FAILED: ", throwable));
     }
 
-    private Mono<Void> logRequest(final ClientLogger logger, final HttpRequest request) {
-        final HttpLogDetailLevel httpLogLevel = httpLogOptions.getLogLevel();
-        if (httpLogLevel.shouldLogUrl()) {
-            UrlBuilder requestUrl = UrlBuilder.parse(request.getUrl());
-            requestUrl.setQuery(getAllowedQueryString(request.getUrl().getQuery()));
-            logger.info("--> {} {}", request.getHttpMethod(), requestUrl.toString());
+    private Mono<String> logRequest(final ClientLogger logger, final HttpRequest request) {
+        /*
+         * Logging is either disabled or the logging level is above information (warning or error), this will result
+         * in nothing being logged so perform a no-op.
+         */
+        int numericLogLevel = LoggingUtil.getEnvironmentLoggingLevel().toNumeric();
+        if (numericLogLevel == LogLevel.DISABLED.toNumeric()
+            || numericLogLevel > LogLevel.INFORMATIONAL.toNumeric()) {
```

Comment:
Can this condition be simplified by just using `numericLogLevel > LogLevel.INFORMATIONAL.toNumeric()` since `LogLevel.DISABLED` is 5?

Reply:
Good catch, changed.